### PR TITLE
close#12, Checking version of firmware on hasseb devices

### PR DESCRIPTION
**Pull request by kbacia7**
I added checking version of firmware on hasseb devices with link to github page how-to update firmware

### CODE REVIEW

File: consts.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-"""Constants common the various modules."""
```
Comment (dgomes) [inline consts.py:1]: don't erase...

File: dali-mqtt-daemon.py [hunk 2]
```diff
@@ -88,8 +89,8 @@ def gen_ha_config(light, mqtt_base_topic):
     return json.dumps(json_config)
 
 
-LOG_FORMAT = "%(asctime)s %(levelname)s: %(message)s{}".format(RESET_COLOR)
-logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
```
Comment (dgomes) [inline dali-mqtt-daemon.py:92]: don't touch logging.INFO in this PR

File: dali-mqtt-daemon.py [hunk 3]
```diff
@@ -266,6 +267,7 @@ def create_mqtt_client(
     parser.add_argument(
         "--ha-discover-prefix", help="HA discover mqtt prefix", default="homeassistant"
     )
+    parser.add_argument("--log-level", help="Log level", default="info")
```
Comment (dgomes) [inline dali-mqtt-daemon.py:270]: unrelated to this PR

File: dali-mqtt-daemon.py [hunk 4]
```diff
@@ -289,13 +291,35 @@ def create_mqtt_client(
...
+        "info": logging.INFO,
+        "debug": logging.DEBUG,
+    }
+    args.log_level = args.log_level.lower()
+    if args.log_level not in all_supported_log_levels:
+        logger.error(
+            "Unsupported log level {}! Changed to level info".format(args.log_level)
+        )
+        args.log_level = "info"
+    logger.setLevel(all_supported_log_levels[args.log_level])
```
Comment (dgomes) [inline dali-mqtt-daemon.py:307]: again this is another PR

File: dali-mqtt-daemon.py [hunk 5]
```diff
@@ -289,13 +291,35 @@ def create_mqtt_client(
...
         logger.debug("Using <%s> driver", args.dali_driver)
         if args.dali_driver == HASSEB:
             from dali.driver.hasseb import SyncHassebDALIUSBDriver
 
             dali_driver = SyncHassebDALIUSBDriver()
+            firmware_version = float(dali_driver.readFirmwareVersion())
+            if firmware_version < MIN_HASSEB_FIRMWARE_VERSION:
+                logger.error("Using a script requires newest hasseb firmware")
+                logger.error(
+                    "Please, look at https://github.com/hasseb/python-dali/tree/3dbf4af3b3770431e7351057ea328b4dbcc3a355/dali/driver/hasseb_firmware"
```
Comment (dgomes) [inline dali-mqtt-daemon.py:319]: this url points to a very specific commit (not a good idea)

We should get in touch with hasseb and propose to move this information into his fork Wiki

File: dali-mqtt-daemon.py [hunk 6]
```diff
@@ -341,4 +365,4 @@ def create_mqtt_client(
                         config, outfile, default_flow_style=False, allow_unicode=True
                     )
             except Exception as err:
-                logger.error("Could not save configuration: %s", err)
+                logger.error("Co
```
Comment (dgomes) [inline dali-mqtt-daemon.py:368]: this is clearly a mistake...

**Review by dgomes — CHANGES_REQUESTED** [review]

**Comment by kbacia7** [comment]
Ok. I don't have idea what I did here. I close this PR and prepare another, sorry

**Pull request closed without merging**